### PR TITLE
UserspaceEmulator: Handle PerfEvent syscalls

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
We only froward String setting and FlagPost creation for now, due to the
other performance events being nonsensical to forward

We also prepend `UE: ` to every string submitted to the profiler.

If other PerfEvents are worth forwarding please tell me

### CODE REVIEW

**Comment by Hendiadyoin1** [comment]
Quick question, shall these append data to the profile dump the UserspaceEmulator apparently can do?

File: Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp [hunk 1]
```diff
@@ -283,6 +285,26 @@ int Emulator::virt$profiling_disable(pid_t pid)
     return syscall(SC_profiling_disable, pid);
 }
 
+FlatPtr Emulator::virt$perf_event(int event, FlatPtr arg1, FlatPtr arg2)
+{
+    if (event == PERF_EVENT_SIGNPOST)
+        return syscall(SC_perf_event, PERF_EVENT_SIGNPOST, arg1, arg2);
+    return 0;
```
Comment (bgianfo) [inline Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp:292]: Should we return `-ENOSYS` in these cases?

**Review by bgianfo — COMMENTED** [review]
> Quick question, shall these append data to the profile dump the UserspaceEmulator apparently can do?

I don't have strong feelings either way about this, it makes sense for the current instruction profiling support as you get awesome full fidelity traces. Is there anything else we would get out of a dedicated re-implementation in `UserspaceEmulator`?

File: Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp (hunk 1 above)
Comment (Hendiadyoin1) [reply Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp]: Sure!

**Comment by Hendiadyoin1** [comment]
> > Quick question, shall these append data to the profile dump the UserspaceEmulator apparently can do?
> 
> I don't have strong feelings either way about this, it makes sense for the current instruction profiling support as you get awesome full fidelity traces. Is there anything else we would get out of a dedicated re-implementation in `UserspaceEmulator`?

Not sure if I have done it in the best way, but "It should work"
haven't tested yet

**Comment by Hendiadyoin1** [comment]
So every thing *should* work now
I couldn't really test it due to test-js not grace-fully ending when left to its own devices (Some FPU error (likely my fault)) and killing it through signal interrupt for some reason messes with the currently writing signpost in a way, that the current entry is not written to completion, resulting in an invalid profile.
In addition to that I sadly do not know of another program that uses signposts at all.

**Pull request merged**